sandbox/rocky/tf/algos/ddpg/ddpg.py

```diff
@@ -0,0 +1,233 @@
+        reward = reward.reshape(-1, 1)
+        terminal = terminal.reshape(-1, 1)
+
+        target_action = self._target_actor.predict(next_state)
+        target_Q = self._target_critic.predict_target_Q(
+            next_state, target_action, reward, terminal)
+
+        self._critic_net.train(state, action, target_Q)
+        self._actor_net.train(state)
+
+        self._actor_net.update_target_net()
+        self._critic_net.update_target_net()
+        return
+
+    def _report_total_reward(self, reward, step):
+        summary = tf.Summary()
+        summary.value.add(tag='rollout/reward', simple_value=float(reward))
+        summary.value.add(
+            tag='train/episode_reward', simple_value=float(reward))
+        if self._summary_writer:
+            self._summary_writer.add_summary(summary, step)
+
+    def predict(self, state):
+        action = self._actor_net.predict(np.array(state).reshape(1, -1))[0]
+        if self._action_noise:
+            noise = self._action_noise.gen()
+            action = action + noise
+        action = np.clip(action, self._action_range[0], self._action_range[1])
+        return action
+
+    def train(self,
+              epochs=500,
+              epoch_cycles=20,
+              rollout_steps=100,
+              train_steps=50):
+        state = self._env.reset()
+        if self._action_noise:
+            self._action_noise.reset()
+        total_reward = 0.0
+        episode_step = self._session.run(self._global_step)
+
+        for epoch in range(epochs):
+            for step in range(epoch_cycles):
+                for rollout in range(rollout_steps):
+                    if self._plot:
+                        self._env.render()
+                    action = self.predict(state)
+
+                    next_state, reward, terminal, info = self._env.step(
+                        action * self._max_action)
+                    self._replay_buffer.add_data(state, action,
+                                                 reward * self._reward_scale,
+                                                 terminal, next_state)
+                    state = next_state
+                    total_reward += reward
+                    if terminal:
+                        self._report_total_reward(total_reward, episode_step)
+                        print("epoch %d, total reward %lf\n" % (episode_step,
+                                                                total_reward))
+                        episode_step = self._session.run(
+                            self._global_step.assign_add(1))
+                        total_reward = 0
+                        state = self._env.reset()
+                        if self._action_noise:
+                            self._action_noise.reset()
+
+                for train in range(train_steps):
+                    self._train_net()
+            self.save_session(episode_step)
+
+    def load_session(self):
```

Comment:
We don't need this code if everything is serializable. Make sure you can run your implementation with `run_experiment_lite`.